subsys/llext/llext_mem.c

```diff
@@ -343,13 +348,36 @@ int llext_add_domain(struct llext *ext, struct k_mem_domain *domain)
 #endif
 }
 
+int llext_heap_init_harvard(void *instr_mem, size_t instr_bytes, void *data_mem, size_t data_bytes)
+{
+#ifdef CONFIG_LLEXT_HEAP_DYNAMIC
+	if (llext_heap_inited) {
+		return -EEXIST;
+	}
+#ifdef CONFIG_HARVARD
+	k_heap_init(&llext_instr_heap, instr_mem, instr_bytes);
+	k_heap_init(&llext_data_heap, data_mem, data_bytes);
+#else
+	return -ENOSYS;
+#endif
+	llext_heap_inited = true;
+	return 0;
+#else
+	return -ENOSYS;
+#endif
```

Comment:
```
#if !defined(CONFIG_LLEXT_HEAP_DYNAMIC) || !defined(CONFIG_HARVARD)
	return -ENOSYS;
#else
	if (llext_heap_inited) {
		return -EEXIST;
	}

	k_heap_init(&llext_instr_heap, instr_mem, instr_bytes);
	k_heap_init(&llext_data_heap, data_mem, data_bytes);

	llext_heap_inited = true;
	return 0;
#endif
```
seems easier to me even if it changes the resulting error code in a (still invalid) call sequence
```
llext_heap_init();
llext_heap_init_harvard();
```

Reply:
Changed! Looks a lot cleaner, thanks